### ISSUE TITLE
A type with static methods with the same name as a module generates incorrect error message

### ISSUE THREAD

**Issue by huonw** [rust-lang/rust#14564]
```rust
mod Foo { fn x() {} }

struct Foo;

impl Foo { fn y() {} }

fn main() {}
```

```
<anon>:5:1: 5:23 error: duplicate definition of module `Foo`
<anon>:5 impl Foo { fn y() {} }
         ^~~~~~~~~~~~~~~~~~~~~~
<anon>:3:1: 3:12 note: first definition of module `Foo` here
<anon>:3 struct Foo;
         ^~~~~~~~~~~
error: aborting due to previous error
```

The "first definition" error should be pointing to the `mod Foo`.

**Comment by tamird** [comment]
This no longer errors at all. Compiling @lht's code:
```shell
$ rustc main.rs
main.rs:1:1: 1:12 warning: module `Foo` should have a snake case name such as `foo`, #[warn(non_snake_case)] on by default
main.rs:1 mod Foo { }
          ^~~~~~~~~~~
main.rs:2:1: 2:12 warning: struct is never used: `Foo`, #[warn(dead_code)] on by default
main.rs:2 struct Foo;
          ^~~~~~~~~~~
```

**Comment by jooert** [comment]
Test added in d0e5948, I did reference the wrong issue in the commit message, can be closed now.

**Comment by Aatch** [comment]
Fixed a while it seems.

**Comment by pnkfelix** [comment]
cc #21546 #29185 #26421 ;)
